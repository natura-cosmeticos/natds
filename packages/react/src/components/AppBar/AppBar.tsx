import React from 'react'
import styles from './AppBar.styles'
import type { AppBarProps } from './AppBar.props'

const AppBar = React.forwardRef<HTMLDivElement, AppBarProps>(({
  className = '',
  color = 'default',
  elevation = true,
  testID,
  children
},
ref) => {
  const { appBar } = styles({ elevation, color })

  return (
    <div className={`${className} ${appBar}`} data-testid={testID} ref={ref}>
      {children}
    </div>
  )
})

export default AppBar
