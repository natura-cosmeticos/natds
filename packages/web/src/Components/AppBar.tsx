import React, { FunctionComponent, forwardRef } from "react";
import MaterialAppBar, { AppBarProps } from "@material-ui/core/AppBar";
import { makeStyles, withTheme } from "@material-ui/core/styles";
import { IThemeWeb } from "../Themes";
import { getDefaultTheme } from "./shared";

export interface IAppBarProps extends AppBarProps {

  /**
   * @optional
   */
  theme: IThemeWeb | unknown;
}

const defaultAppBarElevation = 2;

export const AppBar: FunctionComponent<IAppBarProps> = forwardRef((
  props: Omit<IAppBarProps, "translate">,
  ref: any,
) => {
  const {
    elevation,
    theme: providerTheme,
    classes,
  } = props;

  const theme: any = React.useMemo(() => getDefaultTheme(providerTheme), [providerTheme]);

  const useStyles = React.useMemo(() => makeStyles({
    colorDefault: {
      backgroundColor: theme.palette.background.paper,
      color: theme.palette.background.paperContrastText,
    },
  }), [theme]);

  const customClasses = useStyles();

  /**
   * @todo refactor(web): replace elevation magic number by corresponding token
   */
  const NO_ELEVATION = 0;

  return <MaterialAppBar
    {...props}
    classes={{
      ...customClasses,
      ...classes,
    }}
    elevation={!elevation && elevation !== NO_ELEVATION ? defaultAppBarElevation : elevation}
    ref={ref}
  />;
});

export default withTheme(AppBar);
