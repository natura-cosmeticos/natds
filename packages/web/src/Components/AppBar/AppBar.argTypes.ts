import { IAppBarProps } from './AppBar.props'
import { colors } from './__fixtures__/colors'
import { positions } from './__fixtures__/positions'

export const argTypes : Record<keyof IAppBarProps | string, unknown> = {
  children: {
    control: {
      type: 'text'
    },
    description: 'The content of the component.',
    name: 'children',
    table: {
      type: {
        summary: 'node'
      }
    }
  },
  classes: {
    description: 'Override or extend the styles applied to the component. See CSS API for more details.',
    table: {
      type: {
        summary: 'object'
      }
    },
    type: {
      name: 'object',
      required: false
    }
  },
  color: {
    control: {
      options: colors,
      type: 'inline-radio'
    },
    description: 'The color of the component. It supports those theme colors that make sense for this component.',
    table: {
      type: {
        summary: "'default' | 'inherit' | 'primary' | 'secondary'"
      }
    }
  },
  position: {
    control: {
      options: positions,
      type: 'select'
    },
    description: `
The positioning type. The behavior of the different options is described in the MDN web docs.

Note: sticky is not universally supported and will fall back to static when unavailable.`,
    table: {
      type: {
        summary: "'absolute' | 'fixed' | 'relative' | 'static' | 'sticky' | 'fixed'"
      }
    }
  }
}
