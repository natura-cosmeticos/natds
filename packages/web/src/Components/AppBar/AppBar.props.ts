import { AppBarProps } from '@material-ui/core/AppBar'

export type AppBarPosition = 'fixed' | 'absolute' | 'sticky' | 'static' | 'relative';

export type IAppBarProps = Omit<AppBarProps, 'translate'>;
