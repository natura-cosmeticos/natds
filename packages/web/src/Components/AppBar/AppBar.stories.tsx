import * as React from 'react'
import { Meta, Story } from '@storybook/react'

import { AppBar as component } from './AppBar'
import { Toolbar } from '../Toolbar'
import { IAppBarProps } from './AppBar.props'
import { argTypes } from './AppBar.argTypes'
import { ITypographyProps } from '../Typography'
import {
  Template, TemplateWithIconButtons, TemplateWithTabs, ToolbarTemplate
} from './Template'
import { positions } from './__fixtures__/positions'
import { colors } from './__fixtures__/colors'
import { AppBarHighlight } from '../AppBarHighlight'
import { ForAppBarTitle } from '../Typography/Typography.stories'

const AppBar : React.ForwardRefExoticComponent<IAppBarProps> = React.lazy(() => import('./AppBar'))
const Typography : React.ForwardRefExoticComponent<ITypographyProps> = React.lazy(() => import('../Typography'))

AppBar.displayName = 'AppBar'
Typography.displayName = 'Typography'

export default {
  argTypes,
  component,
  subcomponents: { AppBarHighlight, Toolbar },
  title: 'Components/App Bar'
} as Meta

export const Playground : Story<IAppBarProps> = Template.bind({})
Playground.args = {
  children: 'App Bar text',
  position: positions.static
}
Playground.storyName = 'Playground'

export const PrimaryColor : Story<IAppBarProps> = Template.bind({})
PrimaryColor.args = {
  ...Playground.args,
  children: 'Primary color',
  color: 'primary'
}
PrimaryColor.storyName = 'Primary color'

export const SecondaryColor : Story<IAppBarProps> = Template.bind({})
SecondaryColor.args = {
  ...Playground.args,
  children: 'Secondary color',
  color: 'secondary'
}
SecondaryColor.storyName = 'Secondary color'

export const DefaultColor : Story<IAppBarProps> = Template.bind({})
DefaultColor.args = {
  ...Playground.args,
  children: 'Default color',
  color: 'default'
}
DefaultColor.storyName = 'Default color'

export const InheritColor : Story<IAppBarProps> = Template.bind({})
InheritColor.args = {
  ...Playground.args,
  children: 'Inherit color',
  color: 'inherit'
}
InheritColor.storyName = 'Inherit color'

export const DefaultElevation : Story<IAppBarProps> = Template.bind({})
DefaultElevation.args = {
  ...Playground.args,
  children: 'Default elevation',
  color: 'primary'
}

export const NoElevation : Story<IAppBarProps> = Template.bind({})
NoElevation.args = {
  ...Playground.args,
  children: 'No elevation',
  elevation: 0
}
NoElevation.parameters = {
  docs: {
    description: {
      story: `Sometimes, you may need an AppBar without elevation.
This is very common when you are creating a header for a website, for example.`
    }
  }
}

export const WithTabs : Story<IAppBarProps> = TemplateWithTabs.bind({})
WithTabs.args = {
  color: colors.default,
  position: positions.static
}

export const WithProminentToolbar = ToolbarTemplate.bind({})
WithProminentToolbar.args = {
  children: <AppBarHighlight><Typography {...ForAppBarTitle.args}>AppBar text</Typography></AppBarHighlight>,
  variant: 'prominent'
}
WithProminentToolbar.parameters = {
  docs: {
    description: {
      story: 'When `Toolbar` is used with the `prominent` variant, `AppBarHighlight` component will be automatically positioned.'
    }
  }
}

export const WithIconButtons = TemplateWithIconButtons.bind({})
WithIconButtons.args = {
  children: 'AppBar',
  position: positions.relative
}
