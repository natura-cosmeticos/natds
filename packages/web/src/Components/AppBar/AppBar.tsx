import * as React from 'react'
import MaterialAppBar from '@material-ui/core/AppBar'
import { IAppBarProps } from './AppBar.props'
import { useStyles } from './AppBar.styles'

/**
 * By default, an `AppBar` does not include icons or any additional elements.<br />
 * You are free to add the components you want to assemble your `AppBar`.
 *
 * For more advanced usages, check [Material UI AppBar docs](https://material-ui.com/components/app-bar/).
 *
 * ## Importing
 *
 * ```
 * import { AppBar } from '@naturacosmeticos/natds-web';
 * ```
 *
 * @see https://material-ui.com/components/app-bar/
 *
 */
export const AppBar = React.forwardRef<unknown, IAppBarProps>(
  (
    props: IAppBarProps, ref
  ) => {
    const customClasses = useStyles()
    const { classes, elevation } = props

    const getElevation: () => (number) = () => {
      const NO_ELEVATION = 0
      const DEFAULT_ELEVATION = 2

      if (!elevation && elevation !== NO_ELEVATION) {
        return DEFAULT_ELEVATION
      }

      return elevation
    }

    return (
      <MaterialAppBar
        {...props}
        classes={{
          ...customClasses,
          ...classes
        }}
        elevation={getElevation()}
        ref={ref}
      />
    )
  }
)

AppBar.displayName = 'AppBar'

export default AppBar
