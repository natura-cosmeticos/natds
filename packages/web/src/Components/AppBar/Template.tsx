import { Story } from '@storybook/react'
import * as React from 'react'
import { IAppBarProps } from './AppBar.props'
import { IToolbarProps, Toolbar } from '../Toolbar'
import { AppBar } from './AppBar'
import { Typography } from '../Typography'
import { TabContainer } from '../TabContainer'
import { ForAppBar } from '../TabContainer/TabContainer.stories'
import { TabItem } from '../TabItem'
import {
  WithBrandsProductIconAndLabel,
  WithLoveIconAndLabel,
  WithMakeupIconAndLabel,
  WithOutletIconAndLabel
} from '../TabItem/TabItem.stories'
import { Spacing } from '../Spacing'
import { SmallRightMargin } from '../Spacing/Spacing.margin.stories'
import { IconButton } from '../IconButton'
import { Icon } from '../Icon'
import { Button } from '../Button'
import { InheritColor } from '../Button/Button.stories'
import { positions } from './__fixtures__/positions'
import {
  FilterButtonExample,
  MenuButtonExample,
  MyProfileButtonExample,
  SearchButtonExample
} from '../IconButton/IconButton.stories'
import { AppBarHighlight } from '../AppBarHighlight'
import { ForAppBarTitle } from '../Typography/Typography.stories'
import { SmallIcon } from '../Icon/Icon.stories'

export const Template : Story<IAppBarProps> = (args: IAppBarProps) => (
  <AppBar {...args}>
    <Toolbar>
      <AppBarHighlight>
        <Typography {...ForAppBarTitle.args}>{args.children}</Typography>
      </AppBarHighlight>
    </Toolbar>
  </AppBar>
)

export const TemplateWithTabs : Story<IAppBarProps> = (args: IAppBarProps) => {
  const initialState = 0
  const [value, setValue] = React.useState<number>(initialState)

  // eslint-disable-next-line @typescript-eslint/ban-types
  const handleChange = (event: React.ChangeEvent<{}>, newValue: number) => {
    setValue(newValue)
  }

  return (
    <AppBar {...args}>
      {/* @ts-ignore */}
      <TabContainer {...ForAppBar.args} onChange={handleChange} value={value}>
        <TabItem {...WithLoveIconAndLabel.args} />
        <TabItem {...WithOutletIconAndLabel.args} />
        <TabItem {...WithBrandsProductIconAndLabel.args} />
        <TabItem {...WithMakeupIconAndLabel.args} />
      </TabContainer>
    </AppBar>
  )
}

export const ToolbarTemplate : Story<IToolbarProps> = (args: IToolbarProps) => (
  <AppBar position={positions.static}>
    <Toolbar {...args}>
      <Spacing {...SmallRightMargin.args}>
        <IconButton>
          <Icon {...SmallIcon.args} name="filled-brand-naturarosacea" />
        </IconButton>
      </Spacing>
      {args.children}
      <Button {...InheritColor.args}>Login</Button>
    </Toolbar>
  </AppBar>
)

export const TemplateWithIconButtons : Story<IAppBarProps> = (args: IAppBarProps) => (
  <AppBar {...args}>
    <Toolbar>
      <IconButton {...MenuButtonExample.args} />
      <AppBarHighlight>
        <Typography {...ForAppBarTitle.args}>{args.children}</Typography>
      </AppBarHighlight>
      <IconButton {...SearchButtonExample.args} />
      <IconButton {...FilterButtonExample.args} />
      <IconButton {...MyProfileButtonExample.args} />
    </Toolbar>
  </AppBar>
)
