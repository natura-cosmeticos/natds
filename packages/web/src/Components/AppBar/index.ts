import AppBar from './AppBar'

export { IAppBarProps } from './AppBar.props'
export { AppBar }
export default AppBar
