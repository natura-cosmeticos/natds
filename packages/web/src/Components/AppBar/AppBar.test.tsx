import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'

import { AppBar } from './AppBar'

describe('AppBar component', () => {
  describe('when default props are provided', () => {
    it('should render correctly', () => {
      const testRenderer: TestRenderer.ReactTestRenderer = TestRenderer.create(<AppBar />)

      expect(testRenderer).toMatchSnapshot()
    })
  })

  it('when elevation is zero', () => {
    const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(<AppBar elevation={0} />)

    expect(testRenderer).toMatchSnapshot()
  })
})
