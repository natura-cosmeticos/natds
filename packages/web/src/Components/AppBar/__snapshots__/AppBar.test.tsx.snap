// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar component when default props are provided should render correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation2"
/>
`;

exports[`AppBar component when elevation is zero 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation0"
/>
`;
